### ISSUE TITLE
sortBy method in collections loses key

### ISSUE THREAD

**Issue by devngl** [laravel/framework#35505]
- Laravel Version: 8.17.1
- PHP Version: 7.4.12
- Database Driver & Version: MySQL 8

### Description:
A recent change of @taylorotwell  in the sortBy method introduced a breaking change (bug?) that doesn't seem to be documented, or at least I couldn't find it.

I commented it there:
https://github.com/illuminate/collections/commit/8d96c94d98abbd7b5d4366ea6e2a1a8159098699#commitcomment-44841610

### Steps To Reproduce:
TLDR:
I tipically use this structure to fill select inputs:
$query->get()->pluck('name', 'id')->sortBy(['name'])->toArray()

That used to give something like this:
array:1 [▼
1 => "General"
]

Now this is what happens:
array:1 [▼
0 => "General"
]

Regards

**Comment by driesvints** [comment]
As you can see from the type change on the method, an array was previously not supported. So this isn't a breaking change but a new addition.
